product_other_refereces/models/product_template.py

```diff
@@ -0,0 +1,18 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import fields, models, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    other_references = fields.Char()
+
+    @api.model
+    def _name_search(self, name, args, operator="ilike", limit=None, name_get_uid=None):
+        context = self.env.context
+        product_template_extra_domain = ('other_references', operator, name)
+        return super(ProductTemplate, self.with_context(
+            context, product_template_extra_domain=product_template_extra_domain))._name_search(
+                name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid)
```

Comment:
IMO this could be simplified, please check:
```suggestion
        return super(ProductTemplate, self.with_context(
            product_template_extra_domain=product_template_extra_domain))._name_search(
                name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid)
```